### ISSUE TITLE
Agent Finetuning Phase 2 Complete - BC + Reward Training Results

### ISSUE THREAD

**Issue by daniyal-rahman** [daniyal-rahman/ahriuwu#7]
## Training Summary

Agent finetuning (Phase 2) completed successfully with 10 epochs of training.

### Final Metrics

| Metric | Value |
|--------|-------|
| **Top-1 Accuracy** | 48.1% |
| **Top-5 Accuracy** | 86.1% |
| BC Loss | 1.81 |
| Reward Loss | 0.15 |
| Dynamics Loss | 0.0005 |

### Training Configuration

- **Learning Rate**: 3e-4 (constant)
- **Batch Size**: 1
- **Sequence Length**: 32
- **Epochs**: 10
- **Sampler**: 50/50 RewardMixtureSampler (cache-friendly)
- **Dataset**: 146,309 sequences from 70 videos

### Training Curves

![Training Curves](https://raw.githubusercontent.com/daniyal-rahman/ahriuwu/main/docs/images/training_curves.png)

### Key Fixes Applied

1. **RewardHead Zero-Init Bug** (`src/ahriuwu/models/heads.py`)
   - Changed from `nn.init.zeros_(weight)` to `nn.init.normal_(weight, std=0.01)`
   - Zero init caused `output = input @ 0 = 0` always, breaking gradient flow

2. **Cache-Friendly Mixture Sampler** (`src/ahriuwu/data/dataset.py`)
   - Original RewardMixtureSampler caused cache thrashing (0.2 samples/s)
   - New version groups sequences by video while maintaining 50/50 mixture
   - Result: 7.7 samples/s (38x speedup)

3. **Pre-computed Reward Indices**
   - Moved reward index scanning from sampler init to dataset init
   - Prevents slow startup when creating sampler

### Commits

- `b6c56c2` Fix RewardHead zero-init bug + cache-friendly mixture sampler
- `078d3e6` Add action prediction accuracy logging (top-1 and top-5)

### Checkpoints

```
checkpoints/agent_finetune_epoch_001.pt through epoch_010.pt
checkpoints/agent_finetune_latest.pt (= epoch 10)
```

### Next Steps

- [ ] Evaluate on held-out test videos
- [ ] Visualize action predictions vs ground truth
- [ ] Begin Phase 3: Policy learning with imagination

---
*Training time: ~53 hours on single GPU*
